File: LeetCode/python/647.py
def solve(s):
    n=len(s)
    counting=0
    def expend(left,right):
        cnt=0
        while left>=0 and right<n and s[left]==s[right]:
            cnt+=1
            left-=1
            right+=1
        return cnt
    for i in range(n):
        counting+=expend(i,i)
        counting+=expend(i,i+1)
    return counting
s="abc"

print(solve(s))

File: LeetCode/cpp/647.cpp
#include <iostream>
#include <string>
using namespace std;
int countingstring(string s)
{
    int n = s.size();
    int counting = 0;
    auto expand = [&](int left, int right) -> int
    {
        int cnt = 0;
        while (left >= 0 && right < n && s[left] == s[right]) /// 整体加减，中心扩散两边
        {
            cnt++;
            left--;
            right++;
        }
        return cnt;
    };
    for (int i = 0; i < n; i++)
    {
        counting += expand(i, i);
        counting += expand(i, i + 1);
    }
    return counting;
}
int main()
{
    string s = "abc";
    cout << countingstring(s) << endl;
    return 0;
}